servers/src/main/java/tachyon/master/block/BlockMaster.java

```diff
@@ -278,13 +278,17 @@ public void removeBlocks(List<Long> blockIds) {
           if (masterBlockInfo == null) {
             continue;
           }
+          ArrayList<Long> workersToRemove = new ArrayList<Long>();
           for (long workerId : masterBlockInfo.getWorkers()) {
-            masterBlockInfo.removeWorker(workerId);
+            workersToRemove.add(workerId);
```

Comment:
I'm not sure if that's any better than the current solution, I think either of these would need to be changed after we fix up what `MasterBlockInfo` provides. Do you have a strong preference for one implementation?